src/inference/hmc.jl

```diff
@@ -557,40 +426,12 @@ gen_metric(dim::Int, spl::Sampler{<:Hamiltonian}) = AHMC.UnitEuclideanMetric(dim
 gen_metric(dim::Int, ::AHMC.UnitPreconditioner)   = AHMC.UnitEuclideanMetric(dim)
 gen_metric(::Int, pc::AHMC.DiagPreconditioner)    = AHMC.DiagEuclideanMetric(AHMC.getM⁻¹(pc))
 gen_metric(::Int, pc::AHMC.DensePreconditioner)   = AHMC.DenseEuclideanMetric(AHMC.getM⁻¹(pc))
-gen_metric(dim::Int, spl::Sampler{<:AdaptiveHamiltonian}) = gen_metric(dim, spl.info[:adaptor].pc)
+gen_metric(dim::Int, spl::Sampler{<:AdaptiveHamiltonian}) = gen_metric(dim, spl.state.adaptor.pc)
 
 gen_traj(alg::HMC, ϵ) = AHMC.StaticTrajectory(AHMC.Leapfrog(ϵ), alg.n_leapfrog)
 gen_traj(alg::HMCDA, ϵ) = AHMC.HMCDA(AHMC.Leapfrog(ϵ), alg.λ)
 gen_traj(alg::NUTS, ϵ) = AHMC.NUTS(AHMC.Leapfrog(ϵ), alg.max_depth, alg.Δ_max)
 
-function hmc_step(
-    θ,
-    logπ,
-    ∂logπ∂θ,
-    ϵ,
-    alg::T,
-    metric
-) where {T<:Union{HMC,HMCDA,NUTS}}
-    # Make sure the code in AHMC is type stable
-    θ = Vector{Float64}(θ)
-
-    # Build Hamiltonian type and trajectory
-    h = AHMC.Hamiltonian(metric, logπ, ∂logπ∂θ)
-    traj = gen_traj(alg, ϵ)
-
-    h = AHMC.update(h, θ) # Ensure h.metric has the same dim as θ.
-
-    # Sample momentum
-    r = AHMC.rand(h.metric)
-
-    # Build phase point
-    z = AHMC.phasepoint(h, θ, r)
-
-    # Call AHMC to make one MCMC transition
-    z_new, stat = AHMC.transition(traj, h, z)
-
-    return z_new.θ, stat.log_density, stat.is_accept, stat.acceptance_rate
-end
 
 ####
 #### Compiler interface, i.e. tilde operators.
```

Comment:
Why do we have `using Tracker` on L439

Reply:
Great question, I don't think there's a reason. I've pulled it out.